### ISSUE TITLE
TelephoneCallExample: System.ArgumentException : An argument of type System.Int32 is required in position 0

### ISSUE THREAD

**Issue by rosdi** [dotnet-state-machine/stateless#288]
Running the TelephoneCallExample throws out this exception:

```
System.ArgumentException : An argument of type System.Int32 is required in position 0.
   at Stateless.ParameterConversion.Unpack(Object[] args, Type argType, Int32 index)
   at Stateless.ParameterConversion.Unpack[TArg](Object[] args, Int32 index)
   at Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass4_0`1.<ToPackedGuard>b__0(Object[] args)
   at Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass17_0.<UnmetGuardConditions>b__0(GuardCondition c)
   at System.Linq.Enumerable.WhereSelectListIterator`2.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__2(TriggerBehaviour a)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__0(KeyValuePair`2 t)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.ToArray()
   at Stateless.StateMachine`2.ToString()
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object arg0, Object arg1)
   at StatelessTest.PhoneCall.Print() in C:\Repos\StatelessTest\StatelessTest\PhoneCall.cs:line 116
```

This is very similar (but not identical) to issue https://github.com/dotnet-state-machine/stateless/issues/275

**Comment by sgoodgrove** [comment]
Seems to centre around the call

    .InternalTransition<int>(_setVolumeTrigger, (volume, t) => OnSetVolume(volume))

in PhoneCall.cs

`InternalTransition()` in turn makes a call to `InternalTransitionIf()` with an "always true" guard. Inside the call to `InternalTransitionIf()` the guard method is wrapped in a call to `TransitionGuard.ToPackedGuard()` which is where we later get the exception from in the `ToString()` call to `GetPermittedTriggers()`.

I was going to submit a pull request for this (to not wrap the condition), however the behaviour of the various overloads to `InternalTransitionIf()` seem to not behave consistently when it comes to whether `TransitionGuard.ToPackagedGuard()` is used or not, so I thought it would be better left to somebody with more familiarity with the code base.

The following test highlights the issue:

```csharp
        [Fact]
        public void InternalTransitionWithNoGuardWithParamterPermitted()
        {
            var sm = new StateMachine<State, Trigger>(State.A);

            var intTrigger = sm.SetTriggerParameters<int>(Trigger.X);

            sm.Configure(State.A)
                .InternalTransition(intTrigger, (a, b) => { });

            var permittedTriggers = sm.GetPermittedTriggers();

            Assert.Equal(1, permittedTriggers.Count());
        }
```

**Comment by HenningNT** [comment]
Should be fixed in v5.1
